hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java

```diff
@@ -578,6 +591,7 @@ public synchronized TokenIdent cancelToken(Token<TokenIdent> token,
     if (info == null) {
       throw new InvalidToken("Token not found " + formatTokenId(id));
     }
+    removeTokenForOwnerStats(id);
```

Comment:
same here - swap order and update stats as the last step?

Reply:
The reason i left the order as this is that the metric is a in-memory reflection `currentTokens` so it can be calculated once the memory data structure is changed.
The procedure after is for persistent storage.